### ISSUE TITLE
Improve the way how isstudent, isassistant, isteacher and other work

### ISSUE THREAD

**Issue by mudrd8mz** [michael-milette/moodle-filter_filtercodes#3]
Instead of unreliable guessing based on other capabilities, would not it be cleaner if this plugin provided its own capabilities like `filter/filtercodes:isstudent` with a name like "See the content for students". That capability would have the archetype set so that archetype student role gets that capability granted.

Same applies to all other roles.

**Comment by michael-milette** [comment]
Thank you for your feedback @mudrd8mz,

If I understand what you are suggesting, the answer would be no. The {ifstudent}{/ifstudent} can be used on the same page as {ifloggedout}{/ifloggedout} so as to display different messages depending on their role.

For example, lets say you are working with a topic format. You want to make the first topic free but want to have subsequent topics available only to enrolled students. This plugin would allow you to display information about the course to people who are not enrolled and perhaps some instructions and a link to the next topic for those who are enrolled.

If you believe this could be achieved using plugin's own capabilities, please elaborate your thoughts as I do recognize that my approach may not necessarily be the best approach.

**Comment by michael-milette** [comment]
Hi @mudrd8mz,

I noticed that I was using the has_any_capability(). I agree that this seemed quite sub-optimal. My intention was actually to use the has_all_capabilities() function which allows me to identify the role based on multiple criteria instead of any criteria. 

This has now been implemented and hopefully will better meets with your approval.

If not, I am open to alternative approaches if you would be open to sharing suggestions that would fit this particular situation.
